File: api/src/main/java/com/hva/nl/ewa/models/Notification.java
package com.hva.nl.ewa.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity
@Table(name = "notification")
public class Notification implements Model {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private long notificationId;

    @Column
    @NotNull
    private String message;

    @Column
    @CreationTimestamp
    @NotNull
    private Date creationTimestamp;

    @JsonIgnore
    @ManyToOne(targetEntity = Game.class, fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "game_id")
    private Game game;

    @JsonIgnore
    @ManyToOne(targetEntity = User.class, fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    private User user;

    public Notification() {
    }

    public long getNotificationId() {
        return notificationId;
    }

    public void setNotificationId(long notificationId) {
        this.notificationId = notificationId;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Date getCreationTimestamp() {
        return creationTimestamp;
    }

    public void setCreationTimestamp(Date creationTimestamp) {
        this.creationTimestamp = creationTimestamp;
    }

    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: web/src/app/model/Notification.ts
export default class Notification {

  private readonly _notificationId: number;

  private readonly _message: string;

  private readonly _creationTimestamp: string;

  private readonly _sender: string;

  constructor(notificationId: number, message: string, creationTimestamp: string, sender: string) {
    this._notificationId = notificationId;
    this._message = message;
    this._creationTimestamp = creationTimestamp;
    this._sender = sender;
  }

  get notificationId() {
    return this._notificationId;
  }

  get message() {
    return this._message;
  }

  get creationTimestamp() {
    return this._creationTimestamp;
  }

  get sender(): string {
    return this._sender;
  }
}
